### PR TITLE
[full-ci][tests-only] add api tests for lock breaker group

### PR DESCRIPTION
**Pull request by SwikritiT**
## Description
PR https://github.com/owncloud/core/pull/38222 adds a new feature that allows groups of users to be lock breakers. Previously only the owner of file could unlock it. This feature allows members of lock breakers group to unlock a file/folder shared to them.

## Related Issue

- Fixes https://github.com/owncloud/core/issues/38959

## Motivation and Context
 This PR adds api tests to confirm that the feature behaves accordingly.

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiCapabilities-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31094/55/1

File: tests/acceptance/features/apiWebdavLocksUnlock/lockBreakersGroups.feature [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+      | groups |
+      | grp1   |
+      | grp2   |
+    Examples:
+      | dav-path |
+      | old      |
+      | new      |
+
+
+  Scenario Outline: member of the lock breakers groups can unlock a locked folder shared to them
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavLocksUnlock/lockBreakersGroups.feature:36]:
```suggestion
  Scenario Outline: members of the lock breaker groups can unlock a locked folder shared with them
```

File: tests/acceptance/features/apiWebdavLocksUnlock/lockBreakersGroups.feature [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    And 0 locks should be reported for folder "FOLDER" of user "Alice" by the WebDAV API
+    Examples:
+      | dav-path | lock-scope |
+      | old      | shared     |
+      | old      | exclusive  |
+      | new      | shared     |
+      | new      | exclusive  |
+
+
+  Scenario Outline: member of the lock breakers groups can unlock a locked folder shared to them and lock it back again
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavLocksUnlock/lockBreakersGroups.feature:58]:
```suggestion
  Scenario Outline: members of the lock breakers groups can unlock a locked folder shared with them and lock it back again
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 3]
```diff
@@ -569,6 +571,56 @@ public function numberOfLockShouldBeReported($count, $file, $user) {
 		);
 	}
 
+	/**
+	 * @Then group :expectedGroup should exist as lock breaker groups
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:575]:
```suggestion
	 * @Then group :expectedGroup should exist as a lock breaker group
```

**Review by kiranparajuli589 — COMMENTED** [review]
some grammatical improvements

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php [hunk 4]
```diff
@@ -40,7 +40,7 @@ class AppConfigurationContext implements Context {
 	private $featureContext;
 
 	/**
-	 * @When /^the administrator sets parameter "([^"]*)" of app "([^"]*)" to "([^"]*)"$/
+	 * @When /^the administrator sets parameter "([^"]*)" of app "([^"]*)" to ((?:'[^']*')|(?:"[^"]*"))$/
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/AppConfigurationContext.php:43]: imo existing regex should catch pattern like: `... of app "core" to "['one', 'two']"` giving `['one', 'two']` value for last input as string
see https://regex101.com/r/3DwemA/1/

Comment (SwikritiT) [reply tests/acceptance/features/bootstrap/AppConfigurationContext.php:43]: The parameter needs to be in `["one","two"]` format so, I added that.

Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/AppConfigurationContext.php:43]: 🤔 are the double quotes really needed? strange, why? if that is the case you can also replace single quotes to double...

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 5]
```diff
@@ -569,6 +571,56 @@ public function numberOfLockShouldBeReported($count, $file, $user) {
 		);
 	}
 
+	/**
+	 * @Then group :expectedGroup should exist as a lock breaker groups
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:575]:
```suggestion
	 * @Then group :expectedGroup should exist as a lock breaker group
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 6]
```diff
@@ -569,6 +571,56 @@ public function numberOfLockShouldBeReported($count, $file, $user) {
...
+		$responseXml = HttpRequestHelper::getResponseXml($response, __METHOD__)->data->element;
+		$lockbreakergroup = trim(\json_decode(\json_encode($responseXml), true)['value'], '\'[]"');
+		$actualgroup = explode("\",\"", $lockbreakergroup);
+		if (!\in_array($expectedGroup, $actualgroup)) {
+			Assert::fail("could not find group '$expectedGroup' in lock breakers group");
+		}
+	}
+
+	/**
+	 * @Then following group should exist as a lock breaker groups
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:607]:
```suggestion
	 * @Then following group should exist as lock breaker groups
```

**Review by kiranparajuli589 — COMMENTED** [review]
more grammar :)

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php (hunk 4 above)
Comment (SwikritiT) [reply tests/acceptance/features/bootstrap/AppConfigurationContext.php:43]: yes the double quotes are needed. The value in config is set ` "[\"test\",\"Friends\"]"` like this and it won't work as intended if I pass it with single quotes. How do I add it with double quotes? Doing it like this  `"[\"test\"]"` gives me  `has missing steps`

Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/AppConfigurationContext.php:43]: okay...if double quotes are necessary, the regex change is reasonable. or you can let it be, as `... of app "core" to "['one', 'two']"` and later parse the value replacing `'` in context with `"`

**Review by Talank — COMMENTED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Pull request merged**